CMakeLists.txt

```diff
@@ -682,6 +682,22 @@ if (BUILD_SHARED_LIBS)
     if (LLAMA_METAL)
         set_target_properties(llama PROPERTIES RESOURCE "${CMAKE_CURRENT_SOURCE_DIR}/ggml-metal.metal")
     endif()
+
+    # By default, symbols provided by the sublibs that are not used by mainlib (which is all of them in this case)
+    # are not used. This changes that.
+    if (WIN32)
+        set_target_properties(llama PROPERTIES
+        LINK_FLAGS "/WHOLEARCHIVE"
```

Comment:
This flag apparently breaks MSVC unless you also set `/FORCE:MULTIPLE`. I'm not sure why it's spelled "WIN32" here, as I believe MinGW has its own linker.

Reply:
Thanks, it works